lib/handlers/BaseHandler.ts

```diff
@@ -24,15 +27,34 @@ export default class BaseHandler extends EventEmitter {
     return res.end()
   }
 
-  generateUrl(req: http.IncomingMessage, file_id: string) {
+  generateUrl(req: http.IncomingMessage, id: string) {
+    const forwarded = req.headers.Forwarded as string | undefined
+    const path = this.options.path === '/' ? '' : this.options.path
+
+    if (this.options.respectForwardedHeaders) {
+      if (forwarded) {
+        const host = reForwardedHost.exec(forwarded)?.[1]
+        const proto = reForwardedProto.exec(forwarded)?.[1]
+        if (proto && host) {
+          return `${proto}://${host}${path}/${id}`
+        }
+      }
+
+      const forwardHost = req.headers['X-Forwarded-Host']
+      const forwardProto = (req.headers['X-Forwarded-Proto'] ?? 'https') as string
+
+      if (['http', 'https'].includes(forwardProto) && forwardHost) {
```

Comment:
Using this condition, X-Forwarded-Proto will only be respected if X-Forwarded-Host is also set. Could we make it so that each header can also be supplied individually without the other?